scipy/io/tests/test_wavfile.py

```diff
@@ -84,11 +84,18 @@ def test_read_fail():
         fp.close()
 
 
+def test_read_early_eof_with_data():
+    for mmap in [False, True]:
+        with open(datafile('test-44100Hz-le-1ch-4bytes-early-eof.wav'), 'rb') as fp:
+            rate, data = wavfile.read(fp, mmap=mmap)
+
+        del data
+
+
 def test_read_early_eof():
     for mmap in [False, True]:
-        fp = open(datafile('test-44100Hz-le-1ch-4bytes-early-eof.wav'), 'rb')
-        assert_raises(ValueError, wavfile.read, fp, mmap=mmap)
-        fp.close()
+        with open(datafile('test-44100Hz-le-1ch-4bytes-early-eof-no-data.wav'), 'rb') as fp:
+            assert_raises(ValueError, wavfile.read, fp, mmap=mmap)
```

Comment:
A nicer syntax would be:
```
with pytest.raises(ValueError, match='some substring regex to match'):
    wavefile.read(fp, mmap=mmap)
```
It makes it clearer what error is being caught and ensures it's catching the correct one

Reply:
Implemented